sdk/tests/deqp/framework/common/tcuFloat.js

```diff
@@ -596,37 +633,81 @@ tcuFloat.numberToFloat11 = function(value) {
     return tcuFloat.newFloat11(value).bits();
 };
 
-tcuFloat.float11ToNumber = function(float11) {
-    var x = tcuFloat.newDeFloatFromParameters(float11, tcuFloat.description11);
-    return x.getValue();
-};
+tcuFloat.float11ToNumber = (function() {
+    var x = tcuFloat.newDeFloatFromParameters(0, tcuFloat.description11);
+    return function(float11) {
+        x.deFloatParametersNumber(float11);
+        return x.getValue();
+    };
+})();
 
 tcuFloat.numberToFloat10 = function(value) {
     return tcuFloat.newFloat10(value).bits();
 };
 
-tcuFloat.float10ToNumber = function(float10) {
-    var x = tcuFloat.newDeFloatFromParameters(float10, tcuFloat.description10);
-    return x.getValue();
-};
+tcuFloat.float10ToNumber = (function() {
+    var x = tcuFloat.newDeFloatFromParameters(0, tcuFloat.description10);
+    return function(float10) {
+        x.deFloatParametersNumber(float10);
+        return x.getValue();
+    };
+})();
 
-tcuFloat.numberToHalfFloat = function(value) {
-    return tcuFloat.newFloat16(value).bits();
-};
+/**
+ * Converts a Javascript number to a IEEE 754 half-precision float
+ * @param {number} value (64-bit JS float)
+ * @return {number} IEEE 754 value represented as an integer
+ *
+ * Conversion code from http://stackoverflow.com/questions/32633585/how-do-you-convert-to-half-floats-in-javascript
+ */
+tcuFloat.numberToHalfFloat = (function() {
+    var float32View = new Float32Array(1);
+    var int32View = new Int32Array(float32View.buffer);
+
+    return function toHalf( fval ) {
+        float32View[0] = fval;
+        var fbits = int32View[0];
+        var sign  = (fbits >> 16) & 0x8000; // sign only
+        var val   = ( fbits & 0x7fffffff ) + 0x1000; // rounded value
+
+        if( val >= 0x47800000 ) {  // might be or become NaN/Inf
+            if( ( fbits & 0x7fffffff ) >= 0x47800000 ) {
+                // is or must become NaN/Inf
+                if( val < 0x7f800000 ) { // was value but too large
+                    return sign | 0x7c00; // make it +/-Inf
+                }
+                return sign | 0x7c00 | // remains +/-Inf or NaN
+                    ( fbits & 0x007fffff ) >> 13; // keep NaN (and Inf) bits
+            }
+            return sign | 0x7bff; // unrounded not quite Inf
+        }
+        if( val >= 0x38800000 ) { // remains normalized value
+            return sign | val - 0x38000000 >> 13; // exp - 127 + 15
+        }
+        if( val < 0x33000000 )  { // too small for subnormal
+            return sign; // becomes +/-0
+        }
+        val = ( fbits & 0x7fffffff ) >> 23; // tmp exp for subnormal calc
+        return sign | ( ( fbits & 0x7fffff | 0x800000 ) // add subnormal bit
+             + ( 0x800000 >>> val - 102 ) // round depending on cut off
+             >> 126 - val ); // div by 2^(1-(exp-127+15)) and >> 13 | exp=0
+    };
+})();
+// Previously was: return tcuFloat.newFloat16(value).bits();
```

Comment:
Can you format the above copied code a bit?

Let's merge this in for now and get the tests running on bots.

The original code path is still available.  Let's not delete it in case this new path turns out to be buggy in corner cases.